src/Simulation/Simulators/QuantumSimulator/StateDumper.cs

```diff
@@ -155,25 +323,163 @@ public virtual string Format(uint idx, double real, double img)
 
             }
 
+            public override string ToString()
+            {
+                if(Amplitudes == null)
+                {
+                    return "";
+                }
+
+                string retVal = "";
+                uint idx = 0;
+                foreach (Complex complexNum in Amplitudes)
```

Comment:
This may need to be changed to significant amplitudes to allow the new truncation option to propagate out to C# and command-line users.

Reply:
The fragment is already changed to the following:
```c#
            public override string ToString()
            {
                return string.Join('\n', Amplitudes.Select((amplitude, idx) => 
                    Format((uint)idx, amplitude.Real, amplitude.Imaginary)));
            }
```
Based on that what change do you propose? Something like this?
```c#
            public override string ToString()
            {
                return string.Join('\n', 
                    SignificantAmplitudes(BasisStateLabelingConvention.LittleEndian, true, 0.8)  // Fixed settings!!!
                        .Select(
                            item =>
                            {
                                var (cmplx, basisLabel) = item;
                                var amplitude = (cmplx.Real * cmplx.Real) + (cmplx.Imaginary * cmplx.Imaginary);
                                var angle = System.Math.Atan2(cmplx.Imaginary, cmplx.Real);

                                return $"|{basisLabel}⟩\t{FormatCartesian(cmplx.Real, cmplx.Imaginary)}\t == \t" +
                                                    $"{FormatPolar(amplitude, angle)}";
                            }));
            }
```